packages/circuits/helpers/reveal-substring.circom

```diff
@@ -0,0 +1,48 @@
+pragma circom 2.1.6;
+
+include "circomlib/circuits/comparators.circom";
+include "../utils/array.circom";
+
+/// @title RevealSubstring
+/// @notice This circuit reveals a substring from an input array and verifies its uniqueness
+/// @dev Ensures the revealed substring occurs exactly once in the input
+/// @param maxLength The maximum length of the input array
+/// @param maxSubstringLength The maximum length of the substring to be revealed
+template RevealSubstring(maxLength, maxSubstringLength) {
+    assert(maxSubstringLength < maxLength);
+
+    signal input in[maxLength];
+    signal input substringStartIndex;
+    signal input substringLength;
+
+    signal output substring[maxSubstringLength];
+
+    // substringStartIndex should be less than maxLength
```

Comment:
Is this really necessary given the check on `substringStartIndex + substringLength`? Additionally we have been assuming valid index on util circuits, and asks the consuming circuit to do these range checks. This is to avoid redundant checks when main circuit use many of these utils.
Though I think its ok to do `substringStartIndex + substringLength < maxLength` in here.

We can also assert substringLength is not Zero?

Reply:
Yeah, makes sense. I'll add a note above the template regarding this 🙌